[T7][W10-C3]

filbert-c opened the pull request:
Previous commits:
Refactoring for V0.1

Seeking assistance on the issues below

1.Created fixed size stack intended for use only by Command Type class
2.Added UndoRedo Manager
3.Added unexecuted commands for undoable commands (stub for delete)
4.Added undo command
## Seeking opinions on

> use of System.lineSeparator() in UndoCommand.java at seedu.address.logic.commands
> calling execute in sibling class in UndoCommand.java
> @suppress(serial), @suppress(unchecked) for creation of own classes with preconditions at seedu.address.logic.undoredomanager

YesYouKenSpace committed:
Updated UserGuide.md

Update UserGuide.md

Update UserGuide.md

YesYouKenSpace committed:
Update UserGuide.md

YesYouKenSpace committed:
Update UserGuide.md

YesYouKenSpace committed:
Merge branch 'master' of https://github.com/nus-cs2103-AY1617S1/addressbook-level4

YesYouKenSpace committed:
refactored AddressBook to ToDo

YesYouKenSpace committed:
Update AboutUs.md

filbert-c committed:
Enhanced UserGuide Add Command
Included diagram for Add Command

filbert-c committed:
Add UserGuide solution for conflicts in add command

Commit:
Added Project Mentor in AboutUs Page.

Commit:
Edited README.md file

Commit:
Edited README.md

Commit:
Minor change in README.md

Commit:
Revert "Minor change in README.md"

This reverts commit 8bdf20ef8039a585971aad89a81bc077abc12408.

filbert-c committed:
Updated userGuide with more info and graphs
- Delete
- Undo , Redo

filbert-c committed:
updated Redo Graph which was wrong

Commit:
Added some changes to UserGuide.md

Commit:
Corrected some error

filbert-c committed:
Formating correction of redo quote

filbert-c committed:
Merge remote-tracking branch 'origin/refactor-addressbook'

YesYouKenSpace committed:
Update DeveloperGuide.md

YesYouKenSpace committed:
Renamed addressBook to toDo

YesYouKenSpace committed:
Renamed testcases

YesYouKenSpace committed:
Merge pull request #9 from CS2103AUG2016-W10-C3/refactor-addressbook

Refactor addressbook

YesYouKenSpace committed:
Refactored seedu.model.person package

YesYouKenSpace committed:
Refactored Person class and references

YesYouKenSpace committed:
Refactored ReadOnlyPerson and references

YesYouKenSpace committed:
Refactored PersonBuilder class

YesYouKenSpace committed:
Refactored TypicalTestPersons-->TypicalTestTasks and Email-->Description

YesYouKenSpace committed:
Fixed some test cases

YesYouKenSpace committed:
Refactored Phone into Time  'This broke everything'

YesYouKenSpace committed:
Fixed test cases

YesYouKenSpace committed:
Fixed test cases

YesYouKenSpace committed:
Allow attributeless task

YesYouKenSpace committed:
Merge branch 'refactor-person-to-task' of https://github.com/CS2103AUG2016-W10-C3/main into refactor-person-to-task

YesYouKenSpace committed:
Merge branch 'refactor-person-to-task'

Commit:
1.Created fixed size stack intended for use only by Command Type class
2.Added UndoRedo Manager
3.Added unexecuted commands for undoable commands (stub for delete)
4.Added undo command

Seeking opinions on
use of System.lineSeparator() in UndoCommand.java at seedu.address.logic.commands
calling execute in sibling class in UndoCommand.java
@suppress(serial), @suppress(unchecked) for creation of own classes with preconditions at seedu.address.logic.undoredomanager

damithc commented on README.md:
```diff
@@ -1,34 +1,17 @@
-#### Acknowledgements
-
-* Some parts of this sample application were inspired by the excellent 
-  [Java FX tutorial](http://code.makery.ch/library/javafx-8-tutorial/) by *Marco Jakob*. 
```
You should keep this section.

damithc commented on docs/AboutUs.md:
```diff
@@ -4,49 +4,35 @@ We are a team based in the [School of Computing, National University of Singapor
-Role: Developer <br>  
-Responsibilities: Dev Ops
+#### [Matthew Chan](https://github.com/MCHowl) 
+<img src="https://avatars0.githubusercontent.com/u/15246571?v=3&s=466" width="150"><br>
```
Should be a real photo

damithc commented on docs/DeveloperGuide.md:
```diff
@@ -9,7 +9,7 @@
 * [Appendix C: Non Functional Requirements](#appendix-c--non-functional-requirements)
 * [Appendix D: Glossary](#appendix-d--glossary)
-* [Appendix E : Product Survey](#appendix-e-product-survey)
+* [Appendix E : Product Survey](#appendix-e--product-survey)
```
Good catch

damithc commented on docs/DeveloperGuide.md:
```diff
@@ -312,5 +314,9 @@ Use case ends.
 
-{TODO: Add a summary of competing products}
+| Existing Product | Pros | Cons |
+| :---: | :---: | :---: |
```
Perhaps you can mention who did the study for each product.

damithc commented on docs/UserGuide.md:
```diff
@@ -1,134 +1,195 @@
+
+## 2. Quick Start
+1. **Launch SmartyDo** Simply double-click on the `SmartyDo.jar` file to start SmartyDo. You will be greeted with a simple interface that has four parts a CalendarBox, a VisualBox, a MessageBox and a command bar. This command bar is where you enter short commands to tell SmartyDo what to do. 
+
```
This kind of description needs visuals to go with it.

damithc commented on docs/UserGuide.md:
```diff
@@ -1,134 +1,195 @@
+
+<img src="images/AddBasicCmd.png" width="500"><br>
+
+- `add TASK_NAME [Priority][Tag][StartTime][EndTime][Recur]`
```
Just one tag?

damithc commented on docs/UserGuide.md:
```diff
@@ -1,134 +1,195 @@
+- `add TASK_NAME [Priority][Tag][StartTime][EndTime][Recur]`
+
+> The application **does not support adding** new `Tasks` which **conflicts** with your current schedule. 
+> However, with priority system you may wish to reschedule the lower priority task to a free time slot.
```
This is too restrictive.

damithc commented on docs/UserGuide.md:
```diff
@@ -1,134 +1,195 @@
+| `mark`	    |
+
+> SmartyDo **does not store** history of actions in your computer.
+> The history of actions are initialized to empty stack when SmartyDo is opened.
```
Should the user be even seeing the word `stack`?

damithc commented:
Some comments added. Please close the PR after reading comments.

filbert-c committed:
1. Refactored Undo and Redo code
2. Added Test cases for Undo
3. Refactored Person to Task in Add and Delete commands
